xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -31,6 +31,8 @@
 #include "utils/CPUInfo.h"
 #include "utils/TimeUtils.h"
 #include "utils/log.h"
+#include "windowing/android/AndroidUtils.h"
+#include "windowing/WinSystem.h"
```

Comment:
Making the codec dependent on 'windowing' feels wrong. What has a codec to do with the window system?

Reply:
It doesn't, I just don't know where else this can be implemented.

Reply:
It should probably be all passed as a stream hint, I guess.

Reply:
> include is required for CServiceBroker::GetWinSystem()->GetDisplayHDRCapabilities().

This information originally comes from the Android specific code we are in here, can’t you get the info directly from that low level layer?

Reply:
@ksooo Do you mean using `CAndroidUtils::GetDisplaySupportedHdrTypes` directly? I guess I could, but it's not as clean.

Reply:
@quietvoid what do you mean with it's an instance function?

Reply:
I might be mistaken, but I think the `CAndroidUtils::GetDisplaySupportedHdrTypes()` function can only be called on an instance of the class?
Although it doesn't use the member variables at all.

This is also why I made the new `CAndroidUtils::SupportsMediaCodecMimeType` function static.

Reply:
I'm not an expert in the area, but if it doesn't use any members, what's the issue with turning it into a static method instead?

Reply:
I think it would work, I just don't know if it's correct.

Reply:
If I look at this class, I don't see any reason why we would need to make any method non-static that does not access one of the members.
@ksooo What's your opinion on this?

Reply:
I've updated the ones I'm using to be static, it seems to work fine.

Reply:
> The WinSystem.h include is required for CServiceBroker::GetWinSystem()->GetDisplayHDRCapabilities()

An alternative way can be:

```c++
  auto winSystem = dynamic_cast<CWinSystemAndroid*>(CServiceBroker::GetWinSystem());
  winSystem->GetDisplayHDRCapabilities()
```

Reply:
@thexai In the latest changes I've just moved the Android method to `CAndroidUtils` so the win system isn't being used.